Mathlib/Data/Sum/Basic.lean

```diff
@@ -0,0 +1,493 @@
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mario Carneiro, Yury G. Kudryashov
+-/
+import Mathlib.Logic.Function.Basic
+
+/-!
+# Disjoint union of types
+
+This file proves basic results about the sum type `α ⊕ β`.
+
+`α ⊕ β` is the type made of a copy of `α` and a copy of `β`. It is also called *disjoint union*.
+
+## Main declarations
+
+* `Sum.get_left`: Retrieves the left content of `x : α ⊕ β` or returns `none` if it's coming from
+  the right.
+* `Sum.get_right`: Retrieves the right content of `x : α ⊕ β` or returns `none` if it's coming from
+  the left.
+* `Sum.is_left`: Returns whether `x : α ⊕ β` comes from the left component or not.
+* `Sum.is_right`: Returns whether `x : α ⊕ β` comes from the right component or not.
+* `Sum.map`: Maps `α ⊕ β` to `γ ⊕ δ` component-wise.
+* `Sum.elim`: Nondependent eliminator/induction principle for `α ⊕ β`.
+* `Sum.swap`: Maps `α ⊕ β` to `β ⊕ α` by swapping components.
+* `Sum.lex`: Lexicographic order on `α ⊕ β` induced by a relation on `α` and a relation on `β`.
+
+## Notes
+
+The definition of `Sum` takes values in `Type*`. This effectively forbids `Prop`- valued sum types.
+To this effect, we have `PSum`, which takes value in `Sort*` and carries a more complicated
+universe signature in consequence. The `Prop` version is `or`.
+-/
+
+
+universe u v w x
+
+variable {α : Type u} {α' : Type w} {β : Type v} {β' : Type x} {γ δ : Type _}
+
+namespace Sum
+
+deriving instance DecidableEq for Sum
+
+@[simp]
+theorem «forall» {p : Sum α β → Prop} : (∀ x, p x) ↔ (∀ a, p (inl a)) ∧ ∀ b, p (inr b) :=
+  ⟨fun h => ⟨fun _ => h _, fun _ => h _⟩, fun ⟨h₁, h₂⟩ => Sum.rec h₁ h₂⟩
+
+@[simp]
+theorem «exists» {p : Sum α β → Prop} : (∃ x, p x) ↔ (∃ a, p (inl a)) ∨ ∃ b, p (inr b) :=
+  ⟨fun h =>
+    match h with
+    | ⟨inl a, h⟩ => Or.inl ⟨a, h⟩
+    | ⟨inr b, h⟩ => Or.inr ⟨b, h⟩,
+    fun h =>
+    match h with
+    | Or.inl ⟨a, h⟩ => ⟨inl a, h⟩
+    | Or.inr ⟨b, h⟩ => ⟨inr b, h⟩⟩
+
+theorem inl_injective : Function.injective (inl : α → Sum α β) := fun _ _ => inl.inj
+
+theorem inr_injective : Function.injective (inr : β → Sum α β) := fun _ _ => inr.inj
+
+section get
+
+/-- Check if a sum is `inl` and if so, retrieve its contents. -/
+@[simp]
+def getLeft : Sum α β → Option α
+  | inl a => some a
+  | inr _ => none
+
+/-- Check if a sum is `inr` and if so, retrieve its contents. -/
+@[simp]
+def getRight : Sum α β → Option β
+  | inr b => some b
+  | inl _ => none
+
+/-- Check if a sum is `inl`. -/
+@[simp]
+def isLeft : Sum α β → Bool
+  | inl _ => true
+  | inr _ => false
+
+/-- Check if a sum is `inr`. -/
+@[simp]
+def isRight : Sum α β → Bool
+  | inl _ => false
+  | inr _ => true
+
+variable {x y : Sum α β}
+
+theorem get_left_eq_none_iff : x.getLeft = none ↔ x.isRight := by
+  cases x <;> simp only [getLeft, isRight, eq_self_iff_true]
+
+theorem get_right_eq_none_iff : x.getRight = none ↔ x.isLeft := by
+  cases x <;> simp only [getRight, isLeft, eq_self_iff_true]
```

Comment:
Perhaps rename and #align these, to use `getLeft` and `getRight`.

Reply:
Actually, I'm uncertain on this sort of renaming, and have asked on [zulip](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/naming.20conventions/near/305785765).

Reply:
Can you check how I've written it and help suggest how to make it correct, please?